server/chat-plugins/scavenger-games.ts

```diff
@@ -269,24 +298,25 @@ const MODES = {
 			},
 
 			onJoin(user) {
-				if (this.room.scavgame.playerlist && !this.room.scavgame.playerlist.includes(user.id)) {
+				if (this.room.scavgame && this.room.scavgame.playerlist && !this.room.scavgame.playerlist.includes(user.id)) {
 					user.sendTo(this.room, 'You are not allowed to join this scavenger hunt.');
 					return true;
 				}
 			},
 
 			onAfterEnd() {
+				if (!this.room.scavgame) return; // should never happen, but typescript is happy
```

Comment:
If it should never happen, consider `throw`ing instead:

```suggestion
				if (!this.room.scavgame) throw new Error("should never happen");
```

It might also make sense to make a temporary variable to access, instead:

```ts
				const game = this.room.scavgame!;
```